### DISCUSSION TITLE
[Request] Patch integration into mod itself (and some other ideas)

### DISCUSSION THREAD

**Discussion by Sharaj00** [FakeFishGames/Barotrauma#9543] [Ideas and suggestions]
i don't want to upload patches for other mods separately from main mod, i want to have ability to integrate patch into it, that if a mod that needs a patch was activated, patch from the folder with main mod was automatically applied
i've seen example of this in Vintage Story and many where else

here some other ideas what i want to see in game:
  - configuration tab for mods (sliders, checkboxes and e.t.c.)
  - game would warn you if some mods override same id to see which mods conflict
  - option to hide limb if parent limb is hidden, and ability to create custom type of limbs (it will fix boobs clipping for uncompatible mods in my mod)
  - ability to display sprite changes(origin, size, sprite itself) instantly.
  - i do not understand why color of hair and beards are separated for npc which results where hair is green and beard is red for example, it looks bad. to fix this, i had to limit color palette for beards only for natural colors
  - i think that players should not see what happens in the sub if they are outside, it would be more immersive if player who is outside could see exterior of submarines instead of x-raying it. yes, this would cause the need to make boats from the inside and outside, but you could make this optional, just add a layer that is displayed on top if the player is outside, this would expand the creative possibilities for submarine creation
  - and yeah, this is more like a dream than really something that devs can implement - seamless level transition, how cool would it be if the company had no loading between levels

_author of [EuropaWaifu++](https://steamcommunity.com/sharedfiles/filedetails/?id=2648774153)_

**Comment by shangjiaxuan** [comment]
Hi, I was making a pr that could patch previously existing items(https://github.com/Regalis11/Barotrauma/pull/9779). This should should make mods that rewrite different parts of the same thing work together. Overriding clothing should be as simple as this (adjusting hidelimb would need other lines though):

```
<override>
    <Item identifier="commonerclothes1">
      <inherit identifier="commonerclothes1">
        <doinherit/>
        <replace sel="./Wearable/sprite/@texture">
          Commoner1.png
        </replace>
      </inherit>
    </Item>
</override>
```
This should work with other mods that also override the clothing, but with more into stats, etc.

A version of lua mod that merged this pr is here: https://steamcommunity.com/sharedfiles/filedetails/?id=2856758496.
Mods re-implemented are here: https://steamcommunity.com/workshop/filedetails/?id=2857596960
Tests and feedback are welcome. Currently only works for items and characters.

**Reply by shangjiaxuan** [reply]
Path reference are automatically updated. Things before `<doinherit/>` references the direct parent xml, after references the parent xml evaluated (after parents inherit grandparents). Path update happens while in `<doinherit/>`, and parent's path gets updated so that the path would be relative to the current xml. Replacing after <doinherit/> will make the path evaluate to current file location's `Commoner1.png`.

**Reply by Sharaj00** [reply]
thats a pretty interesting, thanks for teaching
i'll try to experiment with it in next update

**Reply by shangjiaxuan** [reply]
I'm also a modder in making this pr, and there is no guarantee that this pr will get integrated into the main game by the devs. Still, any test and feedback is welcome (to make sure the pr is usable in ways expected).

**Reply by Sharaj00** [reply]
i have optimized mod for using with your patch, but still have some features what doesn't work
for example, i want to override corpses loadout, but this xml doesn't work
[corpses.txt](https://github.com/Regalis11/Barotrauma/files/11329710/corpses.txt)

**Reply by shangjiaxuan** [reply]
@Sharaj00 currently the mod only supports items, characters and affliction files. Work is underway to add more, but will need to sort out mods as core package first.

**Comment by BiffW** [comment]
>game would warn you if some mods override same id to see which mods conflict

I had created an [issue](https://github.com/Regalis11/Barotrauma/issues/11744#issue-1653557311) similar to this. But the dev says they won't fix it. Though I still think it should be fixed by overriding with warning.
